numpy/lib/financial.py

```diff
@@ -572,8 +581,14 @@ def rate(nper, pmt, pv, fv, when='end', guess=0.10, tol=1e-6, maxiter=100):
     OpenDocument-formula-20090508.odt
 
     """
-    when = _convert_when(when)
+    when = _convert_when(when, sentinel=pmt)
+    # Cast the default floats to decimal to support decimal computations
+    if isinstance(pmt, Decimal):
+        guess = Decimal(guess)
+        tol = Decimal(tol)
```

Comment:
This also promote `float`s which _aren't_ the default, which is incorrect. This should error, IMO:

    rate(nper, pmt_decimal, pv, fv, guess=0.20)

for the same reason that python makes `Decimal(1) + 2.0` error

So maybe change the defaults to `np.NoValue_`, and then set the appropriate floats/Decimals in the function? (and add the defaults to the docstring)